### ISSUE TITLE
Loss of service groups and devices after agent restart

### ISSUE THREAD

**Issue by dberenguer** [telefonicaid/iotagent-ul#392]
**Version 1.10.0**

Every time the UL agent is restarted all service groups and devices are lost. The database still contains the items but the API  only returns empty lists. As result, UL devices can no longer transmit values to the agent (HTTP 404 response).

**Comment by perrolobo** [comment]
That’s not the usual behavior. You may have some configuration, deployment issues that make the agent lose the DB location/configuration when it restarts. 

I recommend to ask for help in stack overflow giving as much detail as possible about your current deployment characteristics.

Anyway we will take a look just in case something unexpected happened.

**Comment by dberenguer** [comment]
Like issue #391 doing IOTA_REGISTRY_TYPE=mongodb fixes this problem. I just confirm that this is the default behavior for  IOTA_REGISTRY_TYPE commented out.

**Comment by fgalan** [comment]
> Like issue #391 doing IOTA_REGISTRY_TYPE=mongodb fixes this problem. I just confirm that this is the default behavior for IOTA_REGISTRY_TYPE commented out.

Yes, that's correct. Please find the following piece of documentation about it (https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/installationguide.md):

> **deviceRegistry** [equivalent to IOTA_REGISTRY_TYPE ]: type of Device Registry to create. Currently, two values are supported: `memory` and `mongodb`. If the former is configured, a transient memory-based device registry will be used to register all the devices. This registry will be emptied whenever the process is restarted. If the latter is selected, a MongoDB database will be used to store all the device information, so it will be persistent from one execution to the other. Mongodb databases must be configured in the mongob section (as described bellow)

All real-world utilization cases use mongodb as registry. Memory registry only makes sense for testing and things like that.

Maybe what could be done in the scope of thi issue is to change de default to mongodb.
